packages/xrpl/src/models/transactions/trustSet.ts

```diff
@@ -89,6 +93,10 @@ export interface TrustSetFlagsInterface extends GlobalFlags {
   tfSetFreeze?: boolean
   /** Unfreeze the trust line. */
   tfClearFreeze?: boolean
+  /** Deep-Freeze the trust line -- disallow sending and recieving the said IssuedCurrency */
+  tfSetDeepFreeze?: boolean
+  /** Clear a Deep-Frozen trust line */
+  tfClearDeepFreeze?: boolean
```

Comment:
could also possibly add a validation check that `tfSetDeepFreeze` and `tfClearDeepFreeze` should not be set in the same transaction? And possibly add a unit test for this. 

unrelated to this PR, but a similar validation can be added for tfSetFreeze and tfClearFreeze to make sure they aren't set at the same time (and a unit test).

Reply:
Hmm, in the worst case, erring users will be greeted with a `temINVALID_FLAG` error here: https://github.com/XRPLF/rippled/blob/1b75dc8bcd536ff09a4d96064725c016e0df7293/src/xrpld/app/tx/detail/SetTrust.cpp#L42 (Users do not lose any transaction fees)

I have elaborated here as well: https://github.com/XRPLF/xrpl.js/pull/2873#discussion_r1936380055

We will need to replicate the `tfTrustSetMask` variable to do a complete validation. It feels a bit of an overkill.